File: Backend/Backend/Models/Database/Entities/Recommendation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Backend.Models.Database.Enum;

namespace Backend.Models.Database.Entities
{
    public class Recommendation
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public string Title { get; set; }

        public string? Description { get; set; }

        public Category? Category { get; set; }

        public string? Location { get; set; }


        public Rating? Rating { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Image> RecommendationImages { get; set; } = new List<Image>();

        [ForeignKey("UserId")]
        public Guid? UserId { get; set; }
        public User? User { get; set; }


    }
}

File: Backend/python-backend/models/recommendation.py
from sqlalchemy import Column, String, DateTime, ForeignKey, Enum as SqlEnum, Numeric
from sqlalchemy.orm import relationship
import uuid
from database import Base
# from .category import Category  # Enum to be defined elsewhere
# from .rating import Rating  # Enum to be defined elsewhere

class Recommendation(Base):
    __tablename__ = "recommendations"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()), index=True)
    title = Column(String, nullable=False)
    description = Column(String)
    category = Column(String)  # Replace with SqlEnum(Category) when enum is defined
    address = Column(String)
    city = Column(String)
    country = Column(String)
    rating = Column(String)  # Replace with SqlEnum(Rating) when enum is defined
    created_at = Column(DateTime)
    tags = Column(String)  # Store as comma-separated string for SQLite compatibility
    user_id = Column(String(36), ForeignKey("users.id"))

    # Relationships
    user = relationship("User", back_populates="recommendations")
    recommendation_images = relationship("Image", back_populates="recommendation") 